src/storage/txn/commands/flashback_to_version_read_phase.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+// #[PerformanceCriticalPath]
+use txn_types::{Key, TimeStamp};
+
+use crate::storage::{
+    mvcc::MvccReader,
+    txn::{
+        commands::{
+            Command, CommandExt, FlashbackToVersion, ProcessResult, ReadCommand, TypedCommand,
+            FLASHBACK_BATCH_SIZE,
+        },
+        sched_pool::tls_collect_keyread_histogram_vec,
+        Result,
+    },
+    ScanMode, Snapshot, Statistics,
+};
+
+command! {
+    FlashbackToVersionReadPhase:
+        cmd_ty => (),
+        display => "kv::command::flashback_to_version_read_phase | {:?}", (ctx),
+        content => {
+            version: TimeStamp,
+            end_key: Option<Key>,
+            next_lock_key: Option<Key>,
+            next_write_key: Option<Key>,
+        }
+}
+
+impl CommandExt for FlashbackToVersionReadPhase {
+    ctx!();
+    tag!(flashback_to_version);
+    request_type!(KvFlashbackToVersion);
+    property!(readonly);
+    gen_lock!(empty);
+
+    fn write_bytes(&self) -> usize {
+        0
+    }
+}
+
+impl<S: Snapshot> ReadCommand<S> for FlashbackToVersionReadPhase {
+    fn process_read(self, snapshot: S, statistics: &mut Statistics) -> Result<ProcessResult> {
+        let mut reader = MvccReader::new_with_ctx(snapshot, Some(ScanMode::Forward), &self.ctx);
+        // Scan the locks.
+        let mut key_locks = Vec::with_capacity(0);
+        let mut has_remain_locks = false;
+        if self.next_lock_key.is_some() {
+            let key_locks_result = reader.scan_locks(
+                self.next_lock_key.as_ref(),
+                self.end_key.as_ref(),
+                // To flashback `CF_LOCK`, we need to delete all locks.
+                |_| true,
+                FLASHBACK_BATCH_SIZE,
+            );
+            statistics.add(&reader.statistics);
+            (key_locks, has_remain_locks) = key_locks_result?;
+        }
+        // Scan the writes.
+        let mut key_writes = Vec::with_capacity(0);
+        let mut has_remain_writes = false;
```

Comment:
`has_remain_writes` will be false if `key_locks.len() >= FLASHBACK_BATCH_SIZE`, though there are still writes to be cleaned.

Reply:
`next_write_key` is set to none and the command is finished after `FlashbackToVersion`.

Reply:
Nice catch! I will fix it and add a test case to cover.